lib/api.js

```diff
@@ -515,6 +521,62 @@ api.declare({
   res.reply(prices);
 });
 
+api.declare({
+  method: 'delete',
```

Comment:
In order to operate well with retries, etc, we should have a separate GET and DELETE endpoint here.  The GET being used to obtain the credential and the DELETE is used to register receipt of the credentials.  This is the same flow we have for provisioner secrets as currently implemented.  Basically, the response handler running to completion without error is not a reliable signal for whether the worker type has received and understood the credential.  We don't want a machine that fails to get its credential be unable to retry and have to shut down.